presto-spi/src/test/java/com/facebook/presto/spi/type/TestTypeSignature.java

```diff
@@ -78,7 +73,102 @@ public void parseRowSignature()
+        assertRowSignature(
+                "row(a decimal(p1,s1),decimal(p2,s2))",
+                ImmutableSet.of("p1", "s1", "p2", "s2"),
+                rowSignature(namedParameter("a", false, decimal("p1", "s1")), unnamedParameter(decimal("p2", "s2"))));
+
+        // named fields of types with spaces
+        assertRowSignature(
+                "row(time time with time zone)",
+                rowSignature(namedParameter("time", false, signature("time with time zone"))));
+        assertRowSignature(
+                "row(time timestamp with time zone)",
+                rowSignature(namedParameter("time", false, signature("timestamp with time zone"))));
+        assertRowSignature(
+                "row(interval interval day to second)",
+                rowSignature(namedParameter("interval", false, signature("interval day to second"))));
+        assertRowSignature(
+                "row(interval interval year to month)",
+                rowSignature(namedParameter("interval", false, signature("interval year to month"))));
+        assertRowSignature(
+                "row(double double precision)",
+                rowSignature(namedParameter("double", false, signature("double precision"))));
+
+        // unnamed fields of types with spaces
+        assertRowSignature(
+                "row(time with time zone)",
+                rowSignature(unnamedParameter(signature("time with time zone"))));
+        assertRowSignature(
+                "row(timestamp with time zone)",
+                rowSignature(unnamedParameter(signature("timestamp with time zone"))));
+        assertRowSignature(
+                "row(interval day to second)",
+                rowSignature(unnamedParameter(signature("interval day to second"))));
+        assertRowSignature(
+                "row(interval year to month)",
+                rowSignature(unnamedParameter(signature("interval year to month"))));
+        assertRowSignature(
+                "row(double precision)",
+                rowSignature(unnamedParameter(signature("double precision"))));
+        assertRowSignature(
+                "row(array(time with time zone))",
+                rowSignature(unnamedParameter(array(signature("time with time zone")))));
+        assertRowSignature(
+                "row(map(timestamp with time zone,interval day to second))",
+                rowSignature(unnamedParameter(map(signature("timestamp with time zone"), signature("interval day to second")))));
+
+        // quoted field names
+        assertRowSignature(
+                "row(\"time with time zone\" time with time zone,\"double\" double)",
+                rowSignature(
+                        namedParameter("time with time zone", true, signature("time with time zone")),
+                        namedParameter("double", true, signature("double"))));
+
+        // allow spaces
+        assertRowSignatureParsable("row( time  time with time zone, array( interval day to seconds ) )",
```

Comment:
You don't need a special assertion method here: `assertRowSignatureParsable`. I believe you can use `assertSignature` here.